src/java.base/share/classes/jdk/internal/util/SystemProps.java

```diff
@@ -74,20 +74,18 @@ public static Map<String, String> initProperties() {
         putIfAbsent(props, "user.dir", raw.propDefault(Raw._user_dir_NDX));
         putIfAbsent(props, "user.name", raw.propDefault(Raw._user_name_NDX));
 
-        // Platform defined encoding cannot be overridden on the command line
+        // Platform defined encodings cannot be overridden on the command line
         put(props, "sun.jnu.encoding", raw.propDefault(Raw._sun_jnu_encoding_NDX));
-        var nativeEncoding = ((raw.propDefault(Raw._file_encoding_NDX) == null)
-                ? raw.propDefault(Raw._sun_jnu_encoding_NDX)
-                : raw.propDefault(Raw._file_encoding_NDX));
+        var nativeEncoding = raw.propDefault(Raw._native_encoding_NDX);
```

Comment:
I'd prefer not see "var" here, only because it's not immediately clear that nativeEncoding is a String.

Reply:
I changed this `var` to `String` and also another occurrence a few lines below.

Reply:
Also, I updated copyrights.